cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java

```diff
@@ -109,18 +114,56 @@ public void getAccessesForRun(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, lineageStore.getRunMetadata(run), SET_METADATA_RECORD_TYPE, GSON);
   }
 
-  private void checkArguments(long start, long end, int levels) throws BadRequestException {
+  private TimeRange checkArguments(String startStr, String endStr, int levels) throws BadRequestException {
```

Comment:
it seems a little weird to have a ``checkArguments`` method that returns a time range. Perhaps we could move the ``levels`` checking out of this method, and have it only be a ``validateAndParseRange()``?